doc/developers/contributing.rst

```diff
@@ -268,100 +253,118 @@ If any of the above seems like magic to you, then look up the `Git documentation
-  large number of samples > 100000, but does not scale in dimensionality:
-  n_features is expected to be lower than 100".
+Before a PR can be merged, it needs to be approved by two core developers.
+Please prefix the title of your pull request with ``[MRG]`` if the
+contribution is complete and should be subjected to a detailed review. An
+incomplete contribution -- where you expect to do more work before receiving
+a full review -- should be prefixed ``[WIP]`` (to indicate a work in
+progress) and changed to ``[MRG]`` when it matures. WIPs may be useful to:
+indicate you are working on something to avoid duplicated work, request
+broad review of functionality or API, or seek collaborators. WIPs often
+benefit from the inclusion of a `task list
+<https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments>`_ in
+the PR description.
+
+In order to ease the reviewing process, we recommend that your contribution
+complies with the following rules before marking a PR as ``[MRG]``. The
+**bolded** ones are especially important:
+
+
+1. **Make sure your code passes the tests**. The whole test suite can be run
+   with `pytest`, but it is usually not recommended since it takes a long
+   time. It is often enough to only run the test related to your changes:
+   for example, if you changed something in
+   `sklearn/linear_model/logistic.py`, running the following commands will
+   usually be enough:
+
+   - `pytest sklearn/linear_model/logistic.py` to make sure the doctest
+     examples are correct
+   - `pytest sklearn/linear_model/tests/test_logistic.py` to run the tests
+     specific to the file
+   - `pytest sklearn/linear_model` to test the whole `linear_model` module
+   - `pytest sklearn/doc/linear_model.rst` to make sure the user guide
+     examples are correct.
+
+   There may be other failing tests, but they will be caught by the CI so
+   you don't need to run the whole test suite locally. Read more in
+   :ref:`testing_coverage`.
+
+2. **Make sure your code is properly commented and documented**, and **make
+   sure the documentation renders properly**. To build the documentation, please
+   refer to our :ref:`contribute_documentation` guidelines. The CI will also
+   build the docs: please refer to :ref:`generated_doc_CI`.
+
+3. **Tests are necessary for enhancements to be
+   accepted**. Bug-fixes or new features should be provided with
+   `non-regression tests
+   <https://en.wikipedia.org/wiki/Non-regression_testing>`_. These tests
+   verify the correct behavior of the fix or feature. In this manner, further
+   modifications on the code base are granted to be consistent with the
+   desired behavior. In the case of bug fixes, at the time of the PR, the
+   non-regression tests should fail for the code base in the master branch
+   and pass for the PR code.
+
+4. **Make sure that your PR does not add PEP8 violations**. Please run
+   `./build_tools/circle/flake8_diff.sh` or `make flake8-diff` on a Unix-like
```

Comment:
I think leaving the `./build_tools/circle/flake8_diff.sh` is safer since we may just move the file and forget to fix this part of the docs?

Reply:
You mean only keeping  `make flake8-diff`?

Reply:
yes